### ISSUE TITLE
Grunt fails to fire at a visible target in range

### ISSUE THREAD

**Issue by kaldamae** [beyond-all-reason/Beyond-All-Reason#1378]
https://www.beyondallreason.info/replays?gameId=cc1b4c635e00533d71f07cff2c7e78eb
In that game at 4:00 majiks' grunt has attack command on a mex and some winds. It fires 2 shots at a mex and then stops, stands around doing nothing for almost 20 seconds and when attacked by enemy pawns, resumes shooting at mex.

There is a video of this uploaded here
https://discord.com/channels/549281623154229250/773285618826346546/1031230757672337440

The grunt had vision of the target.

Expected outcome: grunt fires continuosly until mex is dead, then moves on to next target.

**Comment by Ezahiel** [comment]
yeah tested it and they fire while moving and dont when stop right at the target

**Comment by kaldamae** [comment]
In the same specific spot on the map?

**Comment by Mac6750** [comment]
I recreated this bug with pawn on DSD and similar bug on Titan with other units. Here is link to videos and replays https://drive.google.com/drive/folders/1zfFqEyJY9aCB9Ks9fPFxukPXPmZHwYyB?usp=share_link.

**Comment by KyleAnthonyShepherd** [comment]
Good videos!
2023-02-16 17-21-25.mp4 -> Yup, seems to be a recreation of this specific bug.
2023-02-16 16-22-24.mp4 -> Seems to be caused by friendly fire checks, the mace does not fire due to friendly metal storage in the way. Still kinda a bug in that it does not pathfind around it.
2023-02-16 16-11-48.mp4 -> This appears to be a separate kind of pathfinding bug (I've also seen this when trying to replicate this issue #1378). Where the wreck seems to really confuse the armflea.
